.github/workflows/deploy.yml

```
@@ -0,0 +1,19 @@
+name: Deploy
+
+on:
+  push:
+  merge_group:
+  release:
+    types:
+      - published
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        python-version: [3.8, 3.9, '3.10']
+    uses: qiboteam/workflows/.github/workflows/deploy-pip-poetry.yml@main
```

```suggestion
    uses: qiboteam/workflows/.github/workflows/deploy-pip-poetry.yml@main
    with:
      os: ${{ matrix.os }}
      python-version: ${{ matrix.python-version }}
      publish: ${{ github.event_name == 'release' && github.event.action == 'published' && matrix.os == 'ubuntu-latest' && matrix.python-version == '3.9' }}
    secrets: inherit
```
This is how it is currently done in Qibo.

---

@scarrazza maybe you should add those lines here (if I understood correctly).

---

Yes, indeed, but we are missing the filtering key in the workflow, @Edoardo-Pedicillo could you please have a look?

---

After merging qiboteam/workflow#37 we can submit @andrea-pasquale suggested change.